spec/acceptance/foreman_hostgroups_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'spec_helper_acceptance'
+
+describe 'Scenario: install foreman and manage some hostgroups' do
+  before(:context) { purge_foreman }
+
+  it_behaves_like 'an idempotent resource' do
+    let(:manifest) do
+      <<-PUPPET
+      include foreman
+
+      Foreman_hostgroup {
+        ensure          => present,
+        base_url        => $foreman::foreman_url,
+        consumer_key    => $foreman::oauth_consumer_key,
+        consumer_secret => $foreman::oauth_consumer_secret,
+        ssl_ca          => $foreman::server_ssl_ca,
+        timeout         => 5,
+      }
+
+      foreman_hostgroup { 'example_hostgroup':
+        description => 'An example parent hostgroup',
+      }
+
+      foreman_hostgroup { 'example_hostgroup/child':
+        description => 'An example child hostgroup',
+      }
+
+      include foreman::cli
+      PUPPET
+    end
+  end
+
+  it_behaves_like 'the foreman application'
+  it_behaves_like 'hammer'
+
+  describe command('hammer hostgroup list') do
+    its(:stdout) { is_expected.to include('example_hostgroup/child') }
+  end
+
+  describe 'removing a hostgroup' do
```

Comment:
Is the order defined in this way? I'm never sure with rspec.

Reply:
I think so.  Test runs look correct.

Reply:
I've used https://relishapp.com/rspec/rspec-core/docs/configuration/overriding-global-ordering a few times in the past. May not be a bad thing to be explicit here.